n3fit/src/evolven3fit/evolven3fit_API.py

```diff
@@ -0,0 +1,209 @@
+        dump_evolved_replica(
+            evolved_block, usr_path, int(replica.removeprefix("replica_"))
+        )
+        # fixing_replica_path
+        utils.fix_replica_path(usr_path, int(replica.removeprefix("replica_")))
+    # remove folder
+    os.rmdir(usr_path / "nnfit" / usr_path.stem)
+
+
+def load_fit(usr_path):
+    """
+    Loads all the replica pdfs at fitting scale in usr_path and return the exportgrids
+
+    Parameters
+    ----------
+
+        usr_path: pathlib.Path
+            path to the folder containing the fit  
+    
+    Returns
+    -------
+
+            : dict
+            exportgrids info
+    """
+    nnfitpath = usr_path / "nnfit"
+    replica_list = []
+    for subdir, dirs, files in os.walk(nnfitpath):
+        for dir in dirs:
+            replica_list.append(dir)
+    replica_list.remove("input")
+    # remove the eventual evolution folder
+    try:
+        replica_list.remove(usr_path.stem)
+    except:
+        pass
+    pdf_dict = {}
+    for replica in replica_list:
+        rep_path = pathlib.Path(replica) / (usr_path.stem + ".exportgrid")
+        yaml_file = nnfitpath / rep_path.relative_to(rep_path.anchor)
+        with yaml_file.open() as fp:
+            data = yaml.safe_load(fp)
+        pdf_dict[replica] = data
+    return pdf_dict
+
+
+def construct_eko_cards(usr_path, op_card_dict, t_card_dict):
+    """Return the theory and operator cards used to construct the eko"""
+    # read the runcard
+    my_runcard = utils.read_runcard(usr_path)
+    # theory_card construction
+    theory = Loader().check_theoryID(my_runcard["theory"]["theoryid"]).get_description()
+    theory.pop("FNS")
+    theory.update(t_card_dict)
+    t_card = gen_theory.gen_theory_card(theory["PTO"], theory["Q0"], update=theory)
+    # construct operator card
+    op_x_grid = utils.generate_x_grid()
+    q2_grid = utils.generate_q2grid(theory["Q0"], 1.0e5)
+    op_card = gen_op.gen_op_card(q2_grid, update={"interpolation_xgrid": op_x_grid})
+    op_card.update(op_card_dict)
+    return t_card, op_card
+
+# Temporary solution. Then it will be loaded from the theory itself
+def construct_eko_for_fit(t_card, op_card, save_path=None):
+    """
+    Construct the eko operator needed for evolution of fitted pdfs
+
+    Parameters
+    ----------
+        usr_path: pathlib.Path
+            path to the folder containing the fit  
+        save_path: pathlib.Path
+            path where the eko will be saved (the eko
+            won't be saved if save_path is None)
+    Returns
+    -------
+        : eko.output.Output
+        eko operator
+        : dict
+        theory runcard
+        : dict
+        operator runcard
+    """
+    
+    # generate eko operator (temporary because it will be loaded from theory)
+    eko_op = run_dglap(t_card, op_card)
+    if save_path is not None:
+        try:
+            eko_op.dump_tar(save_path)
+        except:
+            pass
```

Comment:
Since computing the eko is quite expensive, whether the path is or not writable should be check before computing it and if it isn't, fail loudly. I would be very pissed if after spending an hour (or 20 minutes, not sure how long it takes) the operator is not saved (and even worse, with no warnings!)